src/Controls/src/Core/Handlers/Items/iOS/CarouselViewController.cs

```diff
@@ -331,14 +331,28 @@ void CollectionViewUpdating(object sender, NotifyCollectionChangedEventArgs e)
 		[UnconditionalSuppressMessage("Memory", "MEM0003", Justification = "Proven safe in test: MemoryTests.HandlerDoesNotLeak")]
 		void CollectionViewUpdated(object sender, NotifyCollectionChangedEventArgs e)
 		{
+			int targetPosition;
 			if (_positionAfterUpdate == -1)
 			{
 				return;
 			}
 
 			_gotoPosition = -1;
 
-			var targetPosition = _positionAfterUpdate;
+			// We need to update the position while modifying the collection.
```

Comment:
I think we might want for clear read refactor this on this one method ?

Reply:
> I think we might want for clear read refactor this on this one method ?

Hi @rmarinho  , Not clear about your concern.Are you suggesting improvements to the method's implementation itself, or are you referring to enhancing the comments we've added to this method?